### ISSUE TITLE
Ship explosion sound is off from the original

### ISSUE THREAD

**Issue by mike-hartman** [MiSTer-devel/Arcade-Galaga_MiSTer#15]
I have the original PCB and I compared the two on the same setup.  The MISTer version is louder than it should be and it is lacking the bass aspect of the original (it has too much high pitch content).  I don’t notice this issue when running the same rom set in MAME.

Everything else seems great so far playing a few games through 250,000 pts or so.

**Comment by famiclone6502** [comment]
I agree the sample sounds completely off compared to the original. High treble, no bass in the core, compared to very high bass in the original.

**Comment by lagomorph** [comment]
I've noticed this as well and commented on the forum. It almost sounds like it's playing twice as fast. I wonder if the implementation here is running the 54xx at the same clock rate as the CPUs. From some research online it looks like the CPUs are supposed to run at 3072000 Hz and the 54xx at half that or 1536000 Hz. Perhaps this is the issue?

**Comment by MacroFPGA** [comment]
The main clock is half of the CPU clock, but it is then divided by 6 internally in the 54xx (according to mame)

This version of galaga.sv changes the divide from 8 to 12 giving 250Khz for the clock enable.
[galaga.zip](https://github.com/MiSTer-devel/Arcade-Galaga_MiSTer/files/5639076/galaga.zip)

**Comment by blackwine** [comment]
Hi @mike-hartman, hey @famiclone6502. I have implemented low pass filter for the noise generator used here to create explosion sound. It was based on values in the arcade schematic. Please try latest release and tell me how far this is from the real thing. I would be happy if you could record sound in high quality (and video as well if possible), so I could try to reproduce this as close to the real thing as possible.

**Comment by mike-hartman** [comment]
Hi blackwine.  Thank you for working on this.  I have posted super high quality, direct recorded samples of all of the Galaga PCB sounds on the MiSTer forum in this thread:

https://misterfpga.org/viewtopic.php?f=25&t=1613&p=19228#p19228

Also, please consider the following information from a friend who is skilled, but doesn't have time to develop on MiSTer:

_The 54XX custom chip outputs digital white noise, usually that would be a stream of 0's and 1's randomly distributed (https://en.wikipedia.org/wiki/White_noise). However, the 54xx outputs 4bit values, so you can think of that as white noise with a 'volume' of 1-3.

The mistake that many make (including the current MiSTer core) is assuming that data is regular digital audio, and that it can be mixed directly with the output of the Namco sound circuitry. A look at the Galaga schematics reveals that the 54xx output is fed through 3 different single-pole op-amp filters, each with slightly different passive components. The analog result of each op-amp is then summed together to form the 'explosion' sound. The sum of all these parts is that Galaga's code controls the 'volume' of the 54xx white noise to create an explosion with a couple different crescendos, and then fades out gently.

Feeding a stream of digital white noise into an op-amp filter was a common method of generating explosions, bangs, and pops in early 80s arcade hardware. The op-amp acts an active low-pass filter, amplifying the digital noise with a frequency response designed to avoid sharp pops and clicks.

MAME handles the Galaga explosion pretty well. Over the years, they have implemented modeling of various analog sound circuits, including op-amps. Given the complexity, MiSTer's best approach would probably be to use a digital sample. It's definitely possible to do the analog modeling needed in FPGA to emulate op-amp filters, but it is quite specialized work.
